sympy/solvers/solveset.py

```diff
@@ -1117,3 +1118,154 @@ def linsolve(system, *symbols):
     # Return solutions
     solution = FiniteSet(tuple(solution))
     return solution
+
+
+
+###############################################################################
+################################ NLINSOLVE #####################################
+###############################################################################
+
+
+def nlinsolve_matrix(system, symbols):
+    r"""
+    reference : http://people.math.gatech.edu/~aleykin3/math4803spr13/BOOK/chapter1.pdf
```

Comment:
Please use the same documentation style as `linsolve`, first a good description of the method then illustrative examples. Also the references should go the reference section.